validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -216,20 +216,22 @@ public ValidatorApiHandler(
   @Override
   public SafeFuture<Optional<Map<BLSPublicKey, ValidatorStatus>>> getValidatorStatuses(
       List<BLSPublicKey> validatorIdentifiers) {
-    return chainDataProvider
-        .getStateValidators(
-            "head",
-            validatorIdentifiers.stream().map(BLSPublicKey::toString).collect(toList()),
-            new HashSet<>())
-        .thenApply(
-            (maybeList) ->
-                maybeList.map(
-                    list ->
-                        list.stream()
-                            .collect(
-                                toMap(
-                                    ValidatorResponse::getPublicKey,
-                                    ValidatorResponse::getStatus))));
+    return isSyncActive()
+        ? SafeFuture.completedFuture(Optional.empty())
+        : chainDataProvider
+            .getStateValidators(
+                "head",
+                validatorIdentifiers.stream().map(BLSPublicKey::toString).collect(toList()),
+                new HashSet<>())
```

Comment:
I'm very unclear how we got a `ChainDataProvider` in here - that should be part of the REST API implementation and the dependency should go the other way around where the providers call into `ValidatorApiChannel`.  We shouldn't be using the REST API in the internals of Teku like this (ideally that `ValidatorResponse` class shouldn't be accessible at all from here).  Actually same with `ValidatorStatus`, it shouldn't be used from here either.

Probably need to rework this in a follow up PR and get the dependency chain right.

Reply:
I believe I had inserted the ChainDataProvider to ValidatorApiHandler (we create a new one using combined chain data client and recent chain data client in BeaconChainController). The reason was that using ChainDataProvider methods made it easy to draw parallels when implementing the ValidatorApiChannel interface methods for the RemoteValidatorApiHandler and ValidatorApiHandler.

Reply:
I understand your point though and can revert that in another PR.